src/lib/reasoners/bitv.ml

```diff
@@ -227,11 +231,40 @@ module Shostak(X : ALIEN) = struct
       f x, ctx
     let (and+) = (and*)
 
-    let other t sz ctx =
+    let negate_abstract r =
+      List.map (fun { bv = st; sz } ->
+          let st =
+            match st with
+            | Cte b -> Cte (not b)
+            | _ -> assert false
+          in
+          { bv = st; sz }
+        ) r
+
+    let no_variable r =
+      List.for_all (fun { bv = st; _ } ->
+          match st with
+          | Cte _ -> true
+          | Other _ | Ext _ ->
+            (* After normalization, the semantic value in
+               the constructor Other and Ext cannot be a bitvector.
+               Thus, these values are always uninterpreted for the
+               Bitv theory. *)
+            false
+        ) r
```

Comment:
Can we merge these two raising `Failure` if either `Other` or `Ext` are encountered, and catch it when we call `negate_abstract` instead?